File: EventListenerJava/app/src/main/java/com/sampreethamithkumar/eventlistenerjava/MainActivity.java
package com.sampreethamithkumar.eventlistenerjava;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import com.sampreethamithkumar.eventlistenerjava.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
//        setContentView(R.layout.activity_main);

        // inflate() creates an instance of the binding class for the activity to use
        binding = ActivityMainBinding.inflate(getLayoutInflater());

        // Get a reference to the root view
        View view = binding.getRoot();

        // The root view needs to be passed as an input to setContentView()
        setContentView(view);

        // setContentView(R.layout.activity_main);
        //Button reverseButton = findViewById(R.id.reverseBtn);
        binding.reverseBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // EditText editText = findViewById(R.id.editMessage);
                String builder= new StringBuilder(binding.editMessage.getText()).reverse().toString();
                binding.editMessage.setText(builder);
            }
        });
        //Button clearButton= findViewById(R.id.clearBtn);
        binding.clearBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                // EditText editText =findViewById(R.id.editMessage);
                binding.editMessage.setText("");
            }
        });
    }

//        Button reverseButton = findViewById(R.id.reverseBtn);
//
//        reverseButton.setOnClickListener(v -> {
//            EditText editText = findViewById(R.id.editMessage);
//            String reversedString = new StringBuilder(editText.getText()).reverse().toString();
//            editText.setText(reversedString);
//        });
//
//        Button clearButton = findViewById(R.id.clearBtn);
//
//        clearButton.setOnClickListener(v -> {
//            EditText editText =findViewById(R.id.editMessage);
//            editText.setText("");
//        });
//    }
}

File: EventListenerKotlin/app/src/main/java/com/sampreethamithkumar/eventlistenerkotlin/MainActivity.kt
package com.sampreethamithkumar.eventlistenerkotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import com.sampreethamithkumar.eventlistenerkotlin.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    //lateinit is used to initialize a variable later instead of doing it here
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(getLayoutInflater())
        //constant variables declared using val (its value is not supposed to change) like final in java
        val view = binding.root
        setContentView(view)

        binding.reverseBtn.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View?) {
                //Local variables are typically declared and initialized at the same time, in which case the type of the variable is inferred to be the
//                type of the expression you initialize it with, and you do not need to provide it
                var builder = StringBuilder(binding.editMessage.getText()).reverse().toString()
                binding.editMessage.setText(builder)
            }
        })
        binding.clearBtn.setOnClickListener(object : View.OnClickListener {
            override fun onClick(v: View?) {
                binding.editMessage.setText("")
            }
        })
    }
}
}